test/dlc_tests/conftest.py

```diff
@@ -241,40 +241,61 @@ def ec2_public_ip(request):
     return request.param
 
 
-@pytest.fixture(scope="session")
-def region():
-    return os.getenv("AWS_REGION", DEFAULT_REGION)
+@pytest.fixture(scope="function")
```

Comment:
why are we switching to using `function` in all the fixtures?

Reply:
for all usecases, even though just efa needs it? it might increase the boto3 calls and lead to throttling.

Reply:
By making the scope for the `region` fixture `function` instead of `session`, the region can be assigned based on the instance type used for the test, rather than being hardcoded and badly-reassigned like we currently do for p3dn instances. This prevents other fixtures that also need `ec2_client/resource/other regional clients` from erroring-out (or failing silently).

But now, since `region` is a function-scoped fixture, other fixtures that depend on `region` also need to be function-scoped, because a session-scoped fixture cannot be defined based on a function-scoped fixture.

Reply:
lets observe if we see throttling for boto calls in the CI